src/region_growing.jl

```diff
@@ -0,0 +1,137 @@
+
+# An implementation of the Improved Seeded Region Growing Algorithm in Julia
+#
+# Citation:
+# Albert Mehnert, Paul Jackaway (1997), "An improved seeded region growing algorithm"
+# http://www.ee.bgu.ac.il/~itzik/IP5211/Other/Projects/P19_Seeded%20region%20growing.pdf
+
+function srg{CT<:Colorant}(img::AbstractArray{CT,2}, seeds::AbstractVector{Tuple{CartesianIndex{2}, Int}}, diff_fn::Function = (c1,c2)->(sum(abs,c1-c2)))
+
+    # Required data structures
+    result              =   similar(dims->fill(-1,dims), indices(img))  # Array to store labels
+    nhq                 =   Queue(CartesianIndex{2})                                # Neighbours holding queue
+    pq                  =   PriorityQueue(Queue{CartesianIndex{2}}, Float64)        # Priority Queue to hold the queues of same δ value
+    qdict               =   Dict{Float64, Queue{CartesianIndex{2}}}()               # A map to get a reference to queue using the δ value
+    labelsq             =   Queue(Int)                                  # Queue to hold labels
+    holdingq            =   Queue(CartesianIndex{2})                                # Queue to hold points corresponding to the labels in `labelsq`
+    region_means        =   Dict{Int, CT}()                             # A map containing (label, mean) pairs
+    region_pix_count    =   Dict{Int, Int}()                            # A map containing (label, pixel_count) pairs
+
+    # Labelling initial seeds and initialising `region_means` and `region_pix_count`
+    for seed in seeds
+        result[seed[1]] = seed[2]
+        region_pix_count[seed[2]] = get!(region_pix_count, seed[2], 0) + 1
```

Comment:
There can be multiple points having the same label; for eg, the user might set all points of a rectangular region to have a specific label. In that case, we need to keep on updating `region_pix_count`.